### ISSUE TITLE
[no-uncalled-signals] Reporting on boolean operations with called signals

### ISSUE THREAD

**Issue by rdamazio** [angular-eslint/angular-eslint#2495]
**Description and reproduction of the issue**

I'm getting several false positives from this new rule:
https://github.com/rdamazio/efis-editor/actions/runs/15396346129/job/43318065699?pr=230
(unless I'm misunderstanding when it should trigger)

MINIMAL REPRO LINK: https://stackblitz.com/edit/angular-eslint-repros-gh6mo2xr

**Versions**

```sh
# Please run `npx ng version` in your project and paste the full output here:

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 19.0.0
Node: 20.19.1
Package Manager: npm 10.8.2
OS: linux x64

Angular: 19.0.0
... animations, build, cli, common, compiler, compiler-cli, core
... forms, platform-browser, router

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1900.0
@angular-devkit/core         19.0.0
@angular-devkit/schematics   19.0.0
@schematics/angular          19.0.0
rxjs                         7.8.1
typescript                   5.6.3
zone.js                      0.15.0
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
This was already fixed https://github.com/angular-eslint/angular-eslint/commit/aba72541147780537c9a1c4177a5d36a95675438

**Comment by JamesHenry** [comment]
19.7.1 will be out shortly
